sql/catalyst/src/main/scala/org/apache/spark/sql/internal/connector/PredicateUtils.scala

```diff
@@ -132,4 +133,16 @@ private[sql] object PredicateUtils {
       case _ => None
     }
   }
+
+  def toV1(
+      predicates: Array[Predicate],
+      skipIfNotConvertible: Boolean): Array[Filter] = {
```

Comment:
thinking about this more, I think we can just have
```
def toV1(predicates: Array[Predicate]): Array[Filter] = ...
```
In `SupportsDelete`, we can check the number of returned v1 filter
```
  default boolean canDeleteWhere(Predicate[] predicates) {
    Filter[] v1Filters = PredicateUtils.toV1(predicates);
    if (v1Filters.length < predicates) return false;
    return this.canDeleteWhere(v1Filters);
  }


  default void deleteWhere(Predicate[] predicates) {
    this.deleteWhere(PredicateUtils.toV1(predicates));
  }
```

Reply:
I actually have thought about this, this works OK for `SupportsDelete`, but we will have to throw Exception later on for `SupportsOverwrite`, because we don't have an API `canOverwrite`.

Reply:
shall we add `canOverwrite`?

Reply:
at least we can still throw error if we want to, by checking the number of returned v1 filters

Reply:
NVM, I will change to this way, and check `if (v1Filters.length < predicates.length)` and throw Exception in `SupportsOverwrite.overwrite`.

Reply:
Replied without seeing your previous message.

Will add `canOverwrite`